src/rpm/builder.rs

```diff
@@ -537,10 +541,15 @@ impl RPMBuilder {
         header_idx_tag.write(&mut header)?;
         let header = header;
 
-        let (header_digest_sha1, header_and_content_digest_md5) =
-            Self::derive_hashes(header.as_slice(), content.as_slice())?;
-
-        let header_and_content_len = header.len() + content.len();
+        let header_and_content_cursor = SeqCursor::new(&[header.as_slice(), content.as_slice()]);
+        let header_and_content_len = header_and_content_cursor.len();
+        let Digests {
+            header_and_content_digest: header_and_content_digest_md5,
+            header_digest: header_digest_sha1,
+        } = RPMPackage::create_digests_from_readers(
+            &mut header.as_slice(),
+            header_and_content_cursor,
+        )?;
```

Comment:
PAYLOADDIGEST and PAYLOADDIGESTALT, which I'm currently adding support for, can't be generated here, because they are put in the main header rather than the signature header.  So I'll rename this to signify that it's only the digests going in the signature header that are created here.